### PR TITLE
feat(ci): generate and validate SBOM

### PR DESCRIPTION
**Pull request by ba11b0y**
Fixes https://github.com/opensbom-generator/spdx-sbom-generator/issues/298

### CODE REVIEW

File: .github/workflows/build-pr.yml [hunk 1]
```diff
@@ -13,9 +13,9 @@ jobs:
         go: [ "^1.20" ]
     steps:
       - name: Checkout Repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
       - name: Setup Go
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
```
Comment (nishakm) [inline .github/workflows/build-pr.yml:18]: I'm wondering if we need to pin the actions to a specific git commit if it is already tagged with the version number. Also, I wonder how we can watch for later releases.

**Review by nishakm — CHANGES_REQUESTED** [review]
I've got a small question about the version pinning. Other than that, this looks good.

File: .github/workflows/build-pr.yml (hunk 1 above)
Comment (ba11b0y) [reply .github/workflows/build-pr.yml:18]: I remember @puerco commented on another PR in the parsers project mentioning that we wanted commit ID here. I thought that’s a convention being followed.

Comment (ba11b0y) [reply .github/workflows/build-pr.yml:18]: This is the comment
https://github.com/opensbom-generator/parsers/pull/58#discussion_r1209873095

File: Makefile [hunk 2]
```diff
@@ -30,12 +30,15 @@ mod:
 generate: mod
 	@echo "Running cli on version: $(VERSION)"
 	@GO111MODULE=on GLFLAGs=-mod-vendor go run cmd/generator/generator.go $(ARGS)
+	@GO111MODULE=on GLFLAGs=-mod-vendor go run cmd/sbomgen/sbomgen.go $(ARGS) > examples/sbom.spdx
```
Comment (ba11b0y) [inline Makefile:33]: I've added a command alongwith the existing Makefile command to run the refactor to generate a SBOM in the existing examples directory.
This generated SBOM is validated later by the GitHub action.

**Pull request merged**